src/App.jsx

```diff
@@ -7,12 +10,86 @@ import SimulationResults from "./SimulationResults";
 function App() {
   const [builds, setBuilds] = useState([]);
 
+  const addBuild = () => {
+    setBuilds(builds.concat([{ id: uuid() }]));
+  };
+
+  const addRestriction = (buildId) => {
+    const build = _.find(builds, (build) => build.id === buildId);
+    if (_.isNil(build.restrictions)) {
+      build.restrictions = [];
+    }
+    build.restrictions.push({ id: uuid() });
+    setBuilds(_.cloneDeep(builds));
+  };
+
+  const addSetRestriction = (buildId) => {
+    const build = _.find(builds, (build) => build.id === buildId);
+    if (_.isNil(build.setRestrictions)) {
+      build.setRestrictions = [];
+    }
+    build.setRestrictions.push({ id: uuid(), is4Piece: true });
+    updateIs4PieceDisabled(build.setRestrictions);
+    setBuilds(_.cloneDeep(builds));
+  };
+
+  const onCharacterChange = (buildId, character) => {
+    const build = _.find(builds, (build) => build.id === buildId);
+    build.character = character;
+    setBuilds(_.cloneDeep(builds));
+  };
+
+  const onRestrictionRemoval = (buildId, restrictionId) => {
```

Comment:
So take this example: I have a `onRestrictionRemoval` fn that will remove a restriction with the given ids. However, in order for this to work...